### DISCUSSION TITLE
bazel script or ninja make triggers download deps?

### DISCUSSION THREAD

**Discussion by osevan** [dragonflydb/dragonfly#127] [Q&A] [answered]
I want ask you if you download latest boost with script? 

Because it will be better to announce that, because some guys havent internet connection at home. 

So we should download dependency additionaly but manually

**Comment by romange** [comment, marked as answer]
Hey, you should be able to use system installed boost. Where did you see
the boost dependency?

**Reply by osevan** [reply]
Ok, thx

https://defuse.ca/b/35vt1lGn

Cant execute dragonfly

**Reply by romange** [reply]
Can you paste here `/proc/cpuinfo`, @osevan  ?

**Reply by osevan** [reply]
Im running in my arch notebook inside debian vm

**Comment by osevan** [comment]
https://defuse.ca/b/e3erY98M

**Comment by romange** [comment]
wrong paste link

**Reply by osevan** [reply]
Corrected

**Reply by romange** [reply]
it's because your laptop cpu architecture is pretty outdated and is not covered by GCC flags specified by default. 

you should change helio/cmake/internal.cmake by finding the line with haswell and change to `-march=native`. similarly, fix `patches/lua-v5.4.4.patch` by replacing `broadwell` to `native`.

**Reply by osevan** [reply]
Ok thx

**Reply by romange** [reply]
For the record, I probably need to fix the flags. 

```
 vendor_id       : AuthenticAMD
 cpu family      : 22
 model           : 48
 model name      : AMD A8-6410 APU with AMD Radeon R5 Graphics
 stepping        : 1
 microcode       : 0x7030105
 cpu MHz         : 1996.256
 cache size      : 512 KB
 physical id     : 0
 siblings        : 4
 fpu             : yes
 fpu_exception   : yes
 cpuid level     : 13
 wp              : yes
 flags           : fpu vme de pse tsc msr pae mce cx8 apic sep mtrr pge mca cmov pat pse36 clflush mmx fxsr sse sse2 ht syscall nx mmxext fxsr_opt pdpe1gb rdtscp lm rep_good nopl cpuid extd_apicid tsc_known_freq pni pclmulqdq ssse3 cx16 sse4_1 sse4_2 x2apic movbe popcnt tsc_deadline_timer aes xsave avx f16c hypervisor lahf_lm cmp_legacy svm cr8_legacy abm sse4a misalignsse 3dnowprefetch osvw topoext ssbd vmmcall tsc_adjust bmi1 xsaveopt virt_ssbd arat npt nrip_save tsc_scale arch_capabilities
```

**Reply by osevan** [reply]
Compilation runs. ..


Btw with my correct march flag instead native. 

And bazel.sh release stop working when I remove  network card.


Maybe you triggering somewhere install-dependencies.sh?

**Reply by osevan** [reply]
For record you can tell me how I can remove SSL compile with DG. 

Maybe  -Dwithout_ssl ?

**Reply by osevan** [reply]
Btw ninja dragonfly command hangs... when i remove my wifi stick.  And runs back when plugged in

**Reply by osevan** [reply]
We need full build without internet connection dependency

**Reply by osevan** [reply]
When you have telegram,  we are cool linux kernel community with many coders inside.
Try  to join here:
https://t.me/tt_sched

**Reply by romange** [reply]
`bazel.sh -release` indeed downloads thirdparty libs and dependencies but not boost.
I did not provide an option to remove SSL from DF. Feef free to open an issue about it.
Can you please tell me what is the correct march flag for you?

**Reply by romange** [reply]
ninja build indeed downloads thirdpary dependencies. you can see them in `build-opt/third_party` but it's only once.

**Reply by osevan** [reply]
Here my benches :


https://defuse.ca/b/5DtC9jLF

**Reply by osevan** [reply]
> `bazel.sh -release` indeed downloads thirdparty libs and dependencies but not boost. I did not provide an option to remove SSL from DF. Feef free to open an issue about it. Can you please tell me what is the correct march flag for you?

At the moment btver2

But im running automated script to figure out which cpu march flag I should use in my bashscript installs

**Reply by osevan** [reply]
> ninja build indeed downloads thirdpary dependencies. you can see them in `build-opt/third_party` but it's only once.

Once downloaded first time they skip this download  or download again on every install?

**Reply by romange** [reply]
what do you mean install? 

can you pls tell me which `march` option did you use?

**Reply by osevan** [reply]
> what do you mean install?
> 
> can you pls tell me which `march` option did you use?

march=btver2

**Comment by osevan** [comment]
In my home lab keydb is more than 2.5 times faster..

Btw mimaloc isnt the best choice in my eyes you could do build flag option for jemalloc it scales better in my past experiences.

**Reply by romange** [reply]
> In my home lab keydb is more than 2.5 times faster..
> 
> Btw mimaloc isnt the best choice in my eyes you could do build flag option for jemalloc it scales better in my past experiences.

that could be indeed. DF is designed for stronger computers than laptops.

**Reply by osevan** [reply]
> > In my home lab keydb is more than 2.5 times faster..
> > Btw mimaloc isnt the best choice in my eyes you could do build flag option for jemalloc it scales better in my past experiences.
> 
> that could be indeed. DF is designed for stronger computers than laptops.

Than,we should further tweak this and test on other CPUs. ..

**Reply by osevan** [reply]
Here are benches with Moody queue

https://moodycamel.com/blog/2014/a-fast-general-purpose-lock-free-queue-for-c++.htm#benchmarks


And consider maybe you can use newlib as glibc replacement
